Debugger DISK INFO QoL

Michaelangel007 opened the pull request:
The debugger's `DISK INFO` was lacking readability. Small QoL patch.

Repro:

    <F7>
    BPX C65E
    <F7>
    Boot any .woz (floppy) disk
    DISK INFO

Before:
![disk_info_before](https://user-images.githubusercontent.com/7876796/232247671-da4c0530-8965-406a-ab8a-32671780dc0e.png)

After:
![disk_info_after](https://user-images.githubusercontent.com/7876796/232247672-45f7430a-fb7d-4938-9661-8c4e15795969.png)

Also alphabetized the DiskGetCurrent*() helper functions.

Michaelangel007 commented:
@tomcw The last number is the Shift Register.  Would it make more sense to display this with the shift operator as a mnemonic?

    Reading << 03

tomcw commented:
I prefer leaving the text as "bitOffset" (instead of just "Offset"), otherwise it's ambiguous (ie. is it a byte or bit offset?)

tomcw commented:
2. If you look at the strings that `Disk2InterfaceCard::GetCurrentState()` can return, then I think it's only useful to return the shift register for `Writing` and `Reading` states.

3. > Would it make more sense to display this with the shift operator as a visual mnemonic? i.e. `Reading << 03`

Probably fine as-is (so without the "<<" operator).

4. You have done work in another PR to truncate the line at 78 chars. But `Disk2InterfaceCard::GetCurrentState()` can return these longer strings: `Off (spinning)` and `Writing (write protected)`. Worth checking what the line looks like for these 2 cases.

Michaelangel007 commented:
* Split the disk status string into a 2nd line (which I'm not really happy with) but it always guaranteed to fit.
* Colorized the disk status string.
* Added 3 shift symobls:
  * `<<` reading
  * `>>` writing
  * `>|` write protected
* Lastly, mirrored m_shiftReg for non `.WOZ` disks so you can see the last nibble read/wrote.

![image](https://user-images.githubusercontent.com/7876796/234175312-ec10b90c-3d50-49da-9c5d-b6269ec3abcc.png)

Michaelangel007 commented:
@tomcw Made another pass.  I'm pretty happy with this now.  I would have preferred a "short" disk status string to fit on one line but I'll live with a descriptive 2-line output since the colorized output is way nicer to read now.

tomcw commented on source/Disk.cpp:
```diff
@@ -130,24 +132,44 @@ std::string Disk2InterfaceCard::GetCurrentPhaseString(void)
 }
 
-LPCTSTR Disk2InterfaceCard::GetCurrentState(void)
+LPCTSTR Disk2InterfaceCard::GetCurrentState(Disk_Status_e *pDiskState)
```
If you change this function's arg to a reference: `LPCTSTR Disk2InterfaceCard::GetCurrentState(Disk_Status_e& diskState)`, then you can get rid of all the `if(pDiskState)` checks.

Michaelangel007 replied:
Good idea. Fixed.

Michaelangel007 replied:
Switching to a reference allows us to table drive the strings.  See commit [69bb9f9](https://github.com/AppleWin/AppleWin/pull/1212/commits/69bb9f9fff3ed1d3790f0bf17a29d8fdea587d94).

Michaelangel007 commented:
* Refactored output to be cleaner and table driven.
* Added undocumented support for 1 line output.  Will add `CONFIG DISK #` later.

![disk_info_one_line](https://user-images.githubusercontent.com/7876796/234773499-dd6f1157-3658-42ce-bcd8-5c4f99513dd0.png)

Michaelangel007 commented:
Should be good to go now.

tomcw commented on source/Windows/WinFrame.cpp:
```diff
@@ -68,8 +68,9 @@ static const DWORD g_aDiskFullScreenColorsLED[ NUM_DISK_STATUS ] =
-//	RGB(  0,  0,255)  // DISK_STATUS_PROT  -blue-
+	RGB(255,128,  0), // DISK_STATUS_PROT  ORANGE
+	RGB(  0,  0,255), // DISK_STATUS_EMPTY -blue-
+	RGB(  0,1280,128)  // DISK_STATUS_SPIN  -cyan-
```
Just noticed this typo... 1280 should be 128!

Michaelangel007 replied:
Ack!  Thank for the catch, fixed.

tomcw commented:
Thanks Michael. I've merged it into the main line now.
(I meant to remove some of the minor commit comments, but hit the "Commit" button accidentally!)
